modules/apps/product-navigation/product-navigation-user-personal-bar-web/src/main/resources/META-INF/resources/view.jsp

```diff
@@ -57,5 +57,103 @@
 		<span class="sign-in text-default" role="presentation">
 			<aui:icon cssClass="sign-in text-default" data="<%= anchorData %>" image="user" label="sign-in" markupView="lexicon" url="<%= themeDisplay.getURLSignIn() %>" />
 		</span>
+
+		<aui:script>
+			(function () {
+				var signInLink = document.querySelector('.sign-in > a');
+
+				if (!signInLink || signInLink.dataset.redirect !== 'false') {
+					return;
+				}
+
+				var signInURL = '<%= themeDisplay.getURLSignIn() %>';
+
+				var modalSignInURL = Liferay.Util.addParams(
+					'windowState=exclusive',
+					signInURL
+				);
+
+				var setModalContent = function (html) {
+					var fragment = document.createRange().createContextualFragment(html);
```

Comment:
May as well move this into the `if`, otherwise the code looks like this:

```js
const thing = prepareSomethingThatMightNotBeUsed();

if (checkSomething()) {
    useThing(thing);
}
```